sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -48,11 +50,9 @@ object ScalaNativePluginInternal {
     nativeClangPP := interceptBuildException(Discover.clangpp().toFile),
     nativeCompileOptions := Discover.compileOptions(),
     nativeLinkingOptions := Discover.linkingOptions(),
-    nativeMode := Option(System.getenv.get("SCALANATIVE_MODE"))
-      .getOrElse(build.Mode.default.name),
+    nativeMode := Discover.mode(),
     nativeLinkStubs := false,
-    nativeGC := Option(System.getenv.get("SCALANATIVE_GC"))
-      .getOrElse(build.GC.default.name),
+    nativeGC := Discover.GC(),
```

Comment:
Yes, but I was trying to use the Boy Scout Rule. Also, I think having them all consistent and in the build tools rather than in the `sbt` plugin is a good change.